bindings/kafka/0.5.0/channel.json

```diff
@@ -0,0 +1,96 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "http://asyncapi.com/bindings/kafka/0.5.0/channel.json",
+  "title": "Channel Schema",
+  "description": "This object contains information about the channel representation in Kafka.",
+  "type": "object",
+  "additionalProperties": false,
+  "patternProperties": {
+    "^x-[\\w\\d\\.\\x2d_]+$": {
+      "$ref": "http://asyncapi.com/definitions/3.0.0/specificationExtension.json"
+    }
+  },
+  "properties": {
+    "topic": {
+      "type": "string",
+      "description": "Kafka topic name if different from channel name."
+    },
+    "partitions": {
+      "type": "integer",
+      "minimum": 1,
+      "description": "Number of partitions configured on this topic."
+    },
+    "replicas": {
+      "type": "integer",
+      "minimum": 1,
+      "description": "Number of replicas configured on this topic."
+    },
+    "topicConfiguration" : {
+      "description": "Topic configuration properties that are relevant for the API.",
+      "type": "object",
+      "additionalProperties": true,
+      "properties": {
+        "cleanup.policy": {
+          "description": "The [`cleanup.policy`](https://kafka.apache.org/documentation/#topicconfigs_cleanup.policy) configuration option.",
+          "type": "array",
+          "items":{
+            "type": "string",
+            "enum": ["compact", "delete"]
+          }
+        },
+        "retention.ms": {
+          "description": "The [`retention.ms`](https://kafka.apache.org/documentation/#topicconfigs_retention.ms) configuration option.",
+          "type": "integer",
+          "minimum": -1            
+        },
+        "retention.bytes": {
+          "description": "The [`retention.bytes`](https://kafka.apache.org/documentation/#topicconfigs_retention.bytes) configuration option.",
+          "type": "integer",
+          "minimum": -1
+        },
+        "delete.retention.ms": {
+          "description": "The [`delete.retention.ms`](https://kafka.apache.org/documentation/#topicconfigs_delete.retention.ms) configuration option.",
+          "type": "integer",
+          "minimum": 0
+        },
+        "max.message.bytes": {
+          "description": "The [`max.message.bytes`](https://kafka.apache.org/documentation/#topicconfigs_max.message.bytes) configuration option.",
+          "type": "integer",
+          "minimum": 0
+        },
+        "schemaValidationForKeyEnabled": {
+          "description": "It shows whether the schema validation for the message key is enabled.",
+          "type": "boolean"
+        },
+        "schemaLookupStrategyForKey": {
+          "description": "The name of the schema lookup strategy for the message key.",
+          "type": "string"
+        },
+        "schemaValidationForValueEnabled": {
+          "description": "It shows whether the schema validation for the message value is enabled.",
+          "type": "boolean"
+        },
+        "schemaLookupStrategyForValue": {
```

Comment:
What is your thinking about having this here compared with where it is currently in the message bindings? Do you think we need it in both places, or are you thinking we'd move it here?

Reply:
The current message binding object has a property called **schemaLookupStrategy** and it seems like this is the strategy for the message key.

In Confluent, you can set different strategies for your message value and your message key and because these configs exist under the [topic configurations](https://docs.confluent.io/platform/current/installation/configuration/topic-configs.html) I thought keeping those values in topicConfiguration object makes sense.

About the message binding object, I guess we can remove the **schemaLookupStrategy** property and start to use the  **schemaLookupStrategyForValue** and **schemaLookupStrategyForKey** properties added to the topic configuration to avoid having the same value in two different places.

Reply:
Any thoughts about this ? @dalelane

Reply:
IMO, we can keep both of them for now because message binding is what you declare for your clients to understand about the message format and topic configuration is what you tell the broker about the channel and its attributes. If it doesn't make sense to keep both then we can remove it in the further versions.

By the way, it seems like I need to update the bindings repo first so we can continue our discussion under this PR -> https://github.com/asyncapi/bindings/pull/238.